docs/dev/limit-in-pagination.md

```diff
@@ -0,0 +1,62 @@
+## Background
+
+Scrolled search is performed by request to `http://localhost:9200/<index>/_search?scroll=<timeout>` endpoint and it has a mandatory field `size`, which defines page size:
+```json
+{
+    "size" : 5
+}
+```
+Regular (non-scrolled/non-paged) search is performed by request to `http://localhost:9200/<index</_search` endpoint and it has not mandatory fields. Parameter `size` though defines maximum number of docs to be returned by search.
+
+## Problem statement
+
+`LIMIT` clause is being converted to `size` by SQL plugin during push down operation.
+Hereby comes the conflict in using `LIMIT` with pagination.
+
+## Solution
+
+Don't do push down for `LIMIT`. `LimitOperator` Physical Plan Tree node will cut off yielding search results with minimal overhead.

```

Comment:
Rather than have an operator to limit the output with post processing why not include it as part of the base class. We can have an optional limit if push down isn't available that can be performed in the `PhysicalPlan` base class prior to the inheriting class's `next()` call. Another option would be to have the limit post-processing performed in the `ResourceMonitorPlan` prior to calling the `delegate` `next()`.